### DISCUSSION TITLE
Comparison with Pydantic

### DISCUSSION THREAD

**Discussion by jaywonchung** [yukinarit/pyserde#409] [Q&A] [answered]
This is an awesome project!

I just wanted to ask how pyserde compares with pydantic, which also provides (de)serialization and type checking. When would I use pyserde over pydantic?

Thanks.

**Comment by TommyDuc** [comment, marked as answer]
Hi there, as someone who tried out both Pydantic and Pyserde I can provide a few answers.

We develop for a Yocto / Bitbake project, which is a build system for building custom Linux images, mainly for embedded systems. The way Yocto builds its stuff is that it requires custom recipes for the specific target being built from sources. Precompiled packages do not work. What made us switch is that Pydantic is built on top of Rust and that was too much of a hassle to make it work because compiling Rust through a python package is not supported easily. Pyserde on the other hand is pure Python which is really easy to integrate.

As for the main differences (from my limited experience):
* Pyserde is pure python as stated above
* Pyserde uses a decorator while Pedantic uses inheritance. While Pydantic has a decorator, its documentation says it's not equivalent. By relying on a decorator Pyserde can recursively convert dataclasses. AFAIK Pydantic doesn't do that since it relies on inheritance to do its magic.
* Pyserde supports multiple serialization format. Pydantic only supports JSON. However Pydantic can generate JSON schemas.

**Reply by jaywonchung** [reply]
Thanks so much for your answer @TommyDuc! That clarifies things significantly.
